add-on/panels/js/panel.jsx

```diff
@@ -777,10 +752,29 @@ loop.panel = (function(_, mozL10n) {
                   dispatcher={this.props.dispatcher}
                   isOpenedRoom={room.roomToken === this.state.openedRoom}
                   key={room.roomToken}
-                  room={room} />
+                  room={room}/>
               );
             }, this)
           }</div>
+        );
+      }
+    },
+
+    render: function() {
+      if (this.state.error) {
+        // XXX Better end user reporting of errors.
+        console.error("RoomList error", this.state.error);
+      }
+
+      if (this.state.pendingInitialRetrieval) {
+        return this._renderLoadingRoomsView();
+      }
+
+      return (
+        <div className="rooms">
+          {this._renderNewRoomButton()}
+          {this._renderRoomsListTitle()}
```

Comment:
I still don't see why these need to be in separate methods. You can do `{this.state.rooms.length ? <h1>{mozL10n..}</h1> : null}`. You could stash the rooms.length in some variable, e.g., haveRooms and do `{haveRooms ? ... : null}` for both the title and rooms list.